src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -907,6 +897,15 @@ private void RegisterTypeDiagnostic(ITypeSymbol causingType, InvocationDiagnosti
                     _typeDiagnostics[causingType] = typeDiags;
                 }
             }
+
+            private void ReportDiagnostic(DiagnosticDescriptor descriptor, Location? location, params object?[]? messageArgs) =>
+                (Diagnostics ??= new()).Add(new DiagnosticInfo
```

Comment:
nit: this sort of expression-excessiveness is a bad thing IMO.  THere's no perf benefit to squeezing everything into an expression.  So best to just break things up for clarity.  For example:

```
Diagnostics ??= new();
Diagnostics.Add(...);
````

Also, is this safe wrt concurrent execution?

Reply:
👍

The entire incremental parsing routine is single-threaded.